common/src/sql/dbinit.sql

```diff
@@ -976,8 +981,9 @@ CREATE UNIQUE INDEX ON omicron.public.disk (
 ) WHERE
     time_deleted IS NULL;
 
-CREATE INDEX ON omicron.public.disk (
-    attach_instance_id
+CREATE UNIQUE INDEX ON omicron.public.disk (
```

Comment:
This is one of the cases mentioned in the PR description. To paginate the disks attached to an instance, the id needs to be here, and then this can be a UNIQUE index.